docs/spfx/toolchain/implement-ci-cd-with-azure-pipelines.md

```diff
@@ -0,0 +1,202 @@
+          displayName: Upload SharePoint package to Site Collection App Catalog
+        - script: o365 spo app deploy --name $(GetSharePointPackage.SpPkgFileName) --appCatalogUrl $(o365_app_catalog_site_url) --scope ${{ parameters.o365cli_app_catalog_scope }}
+          displayName: Deploy SharePoint package
+```
+
+> [!NOTE] 
+> You can find the latest version of this file on the [sample](https://github.com/SharePoint/sp-dev-build-extensions/tree/master/samples/azure-devops-ci-cd-spfx)
+
+## Defining the Pipeline structure
+Now that the build and deploy stages are defined in their respective templates, it needs to be assembled as a Multi-stage pipeline.  
+This document will describe the structure of the pipeline as well as the different environments in use.  
+Copy the following content in a new file at the root of the project called `azure-pipelines.yml`.  
+```YAML
+name: $(TeamProject)_$(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
+resources:
+- repo: self
+
+trigger:
+  branches:
+    include:
+    - master
+    - develop
+
+stages:
+- stage: build
+  displayName: build
+  jobs:
+    - template: ./azure-pipelines-build-template.yml
+      parameters:
+        name: 'buildsolution'
+- stage: 'deployqa'
+  # uncomment if you want deployments to occur only for a specific branch
+  #condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/develop'))
+  jobs:
+    - template: ./azure-pipelines-deploy-template.yml
+      parameters:
+        job_name: deploy_solution
+        target_environment: 'qa'
+        variable_group_name: qa_configuration
+```
+> [!NOTE] 
+> You can find the latest version of this file on the [sample](https://github.com/SharePoint/sp-dev-build-extensions/tree/master/samples/azure-devops-ci-cd-spfx)
+> [!NOTE] 
+> You can define multiple environments and by duplicating the `deployqa` stage and providing different parameters. If you do so, make sure the stage name, the job name, the target environment and the variable group name are unique.
+> [!NOTE] 
+> You can conditionnally deploy to different environments leveraging [conditions](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml)
+
+## Configuring the credentials for the environments
+Secrets should never be committed to a repository for security reasons. The pipeline described in the previous steps makes use of _variable groups_ to keep configuration values secret. The variable groups need to be created for each environment and the name needs to match what is described in the pipeline definition (here `qa_configuration`).  
+To create the variable group, follow these steps:
+1. Logon to Azure DevOps, navigate to your project
+1. Under _Pipelines_ click on _Library_
```

Comment:
Please change:
- menu items to BOLD, not italic. 
- click => select (accessibility reasons)

Reply:
fixed